lib/bolt/cli.rb

```diff
@@ -507,6 +507,7 @@ def run_plan(plan_name, plan_arguments, nodes, options)
     end
 
     def apply_manifest(code, targets, filename = nil, noop = false)
+      Puppet[:tasks] = false
```

Comment:
Was this a bug?

Reply:
Not per se. `:tasks` defaults to false. When we run `bolt apply`, we just leave it alone and it works fine. When we run other things, we call `Puppet::Pal#with_script_compiler`, which sets `:tasks` to true.

But since we now use PAL / run tasks while setting up plugins, we can wind up with `:tasks` set to true when running `bolt apply`. So we have to make sure to explicitly turn it off.

At the same time, it may be a bug that `Puppet::Pal#with_script_compiler` doesn't restore `:tasks` to whatever its value was previously. `Puppet::Pal#with_catalog_compiler` sets it to false and then _does_ restore it afterward.

Reply:
Opened puppetlabs/puppet#7822 to fix the upstream puppet issue